src/Trains.NET.Engine/GamePlay/GameBoard.cs

```diff
@@ -17,6 +17,7 @@ public class GameBoard : IGameBoard
         private readonly ITimer? _gameLoopTimer;
         private readonly ITerrainMap _terrainMap;
         private readonly IGameStorage? _storage;
+        private readonly Train _reservedTrain = new Train();
```

Comment:
This is a "dummy" train, could also refactor takenTracks (line 97) to be `Dictionary<Track, (Train?, float)>`